src/main/java/io/fabric8/maven/docker/service/DockerAccessFactory.java

```diff
@@ -35,7 +35,8 @@ public DockerAccess createDockerAccess(DockerAccessContext dockerAccessContext)
             DockerConnectionDetector dockerConnectionDetector = createDockerConnectionDetector(dockerAccessContext, dockerAccessContext.getLog());
             DockerConnectionDetector.ConnectionParameter connectionParam =
                     dockerConnectionDetector.detectConnectionParameter(dockerAccessContext.getDockerHost(), dockerAccessContext.getCertPath());
-            String version = dockerAccessContext.getMinimalApiVersion() != null ? dockerAccessContext.getMinimalApiVersion() : API_VERSION;
+            String version = dockerAccessContext.getMinimalApiVersion() != null ? dockerAccessContext.getMinimalApiVersion()
+                    : DockerAccessWithHcClient.fetchApiVersionFromServer(connectionParam.getUrl(), connectionParam.getCertPath(), dockerAccessContext.getMaxConnections(), dockerAccessContext.getLog());
             DockerAccess access = new DockerAccessWithHcClient("v" + version, connectionParam.getUrl(),
```

Comment:
Can't we put the Api detection within the constructor itself ? That way we could also verify whether we can reach the Daemon and we would avoid that static method call (as well maybe even the createHttpClientAndUrlBuilder() method)